commcare_connect/opportunity/models.py

```diff
@@ -22,6 +22,14 @@ def __str__(self):
         return self.name
 
 
+class HQApiKey(models.Model):
```

Comment:
Can we link this to an Organization as well. That would make it possible to lookup an API key in instances where the user is not present.

What's the thinking in attaching this to the user? Is it just a 'created_by' field?

Reply:
> Can we link this to an Organization as well

Yeah, it already is through the opportunity, but we can make that more explicit.

> What's the thinking in attaching this to the user? Is it just a 'created_by' field?

Since API keys are owned by people in HQ, it seemed right to link it to that person/account here (if the account is deleted we would probably want to delete api keys tied to that account). That is especially so for api keys attached to users that used hq credentials to sign up. They really own those keys. In a practical sense, to use the api key, you need a username, so this gives us a username (email) to use, again at least while users have signed up with hq credentials.

In the future we could have users specify an hq username when they add the api key, but for now assuming it matches the user creating it seemed fair.